File: src/tasks/cli/arg/registration/Parameter.java
package tasks.cli.arg.registration;

import static java.util.Objects.requireNonNull;

import tasks.cli.arg.CliArguments;

public abstract class Parameter {
  private final String description;
  private final Repeatable repeatable;

  Parameter(String description, Repeatable repeatable) {
    this.description = requireNonNull(description);
    this.repeatable = repeatable;
  }

  public String description() {
    return description;
  }

  public boolean isRepeatable() {
    return repeatable == Repeatable.REPEATABLE;
  }

  public enum Repeatable {
    REPEATABLE,
    NOT_REPEATABLE,
  }
}

File: src/tasks/cli/command/Parameter.kt
package tasks.cli.command

abstract class Parameter internal constructor(private val description: String, private val repeatable: Repeatable) {

  fun description(): String {
    return description
  }

  fun isRepeatable(): Boolean {
    return repeatable == Repeatable.REPEATABLE
  }

  enum class Repeatable {
    REPEATABLE, NOT_REPEATABLE
  }
}